### PR TITLE
Fixes #20184: Ressource not found when creating a draft and comming back to it to save it as a technique

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/20184

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/DataTypes.elm [hunk 1]
```diff
@@ -65,6 +65,7 @@ type alias Technique =
   , elems       : List MethodElem
   , parameters  : List TechniqueParameter
   , resources   : List Resource
+  , internalId  : TechniqueId
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/DataTypes.elm:68]: That should not be within the technique

File: webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/JsonDecoder.elm [hunk 2]
```diff
@@ -100,6 +100,7 @@ decodeTechnique =
     |> required "calls" (list (lazy (\_ -> decodeMethodElem Nothing)))
     |> required "parameter" (list decodeTechniqueParameter)
     |> required "resources" (list decodeResource)
+    |> optional "internalId" (map TechniqueId string) (TechniqueId "")
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/JsonDecoder.elm:103]: This is never send by rudder api

File: webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/JsonEncoder.elm [hunk 3]
```diff
@@ -23,7 +23,10 @@ encodeDraft draft =
 
 encodeTechnique: Technique -> Value
 encodeTechnique technique =
```
[the remark this answers is not in the document]
Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-web/src/main/elm/editor/sources/JsonEncoder.elm:25]: The save technique Action should know which parameters to use / function to call  depending on the mode (Creation/Clone => With internalId, edit => no internalid

**Comment by ElaadF** [comment]
PR updated with a new commit

**Review by VinceMacBuche — APPROVED** [review]

**Pull request merged**